src/loaders/js.ts

```diff
@@ -21,6 +23,19 @@ export const jsLoader: Loader = async (input, { options }) => {
 
   let contents = await input.getContents()
 
+  const declaration: LoaderResult = []
+
+  if (options.emitTypes === true || (options.emitTypes === 'ts' && input.path.endsWith('.ts'))) {
```

Comment:
- Maybe we can use extname for extension 
- Shouldn't we always check extension name? (i didn't get the point of `true` vs `ts` but for vue loadAsset, we are using correct extension)

Reply:
The option is only if we might want to disable emitting types for .js files - but happy to lose that option.

Reply:
We can reconsider approach for dts generation filter once we wanted to add feature. vue loader might send a hint about its original extension that is used in a filter

Reply:
👍 Let me know if you think we any of these changes need to be made in this PR.